commands/history/inspect.go

```diff
@@ -318,44 +572,21 @@ func runInspect(ctx context.Context, dockerCli command.Cli, opts inspectOptions)
 		fmt.Fprintln(dockerCli.Out())
 	}
 
-	if rec.ExternalError != nil {
-		dt, err := content.ReadBlob(ctx, store, ociDesc(rec.ExternalError))
-		if err != nil {
-			return errors.Wrapf(err, "failed to read external error %s", rec.ExternalError.Digest)
-		}
-		var st spb.Status
-		if err := proto.Unmarshal(dt, &st); err != nil {
-			return errors.Wrapf(err, "failed to unmarshal external error %s", rec.ExternalError.Digest)
-		}
-		retErr := grpcerrors.FromGRPC(status.ErrorProto(&st))
-		for _, s := range errdefs.Sources(retErr) {
-			s.Print(dockerCli.Out())
+	if out.Error != nil {
+		if out.Error.Sources != nil {
+			fmt.Fprint(dockerCli.Out(), string(out.Error.Sources))
 		}
-		fmt.Fprintln(dockerCli.Out())
-
-		var ve *errdefs.VertexError
-		if errors.As(retErr, &ve) {
-			dgst, err := digest.Parse(ve.Vertex.Digest)
-			if err != nil {
-				return errors.Wrapf(err, "failed to parse vertex digest %s", ve.Vertex.Digest)
-			}
-			name, logs, err := loadVertexLogs(ctx, c, rec.Ref, dgst, 16)
-			if err != nil {
-				return errors.Wrapf(err, "failed to load vertex logs %s", dgst)
-			}
-			if len(logs) > 0 {
-				fmt.Fprintln(dockerCli.Out(), "Logs:")
-				fmt.Fprintf(dockerCli.Out(), "> => %s:\n", name)
-				for _, l := range logs {
-					fmt.Fprintln(dockerCli.Out(), "> "+l)
-				}
-				fmt.Fprintln(dockerCli.Out())
+		if len(out.Error.Logs) > 0 {
+			fmt.Fprintln(dockerCli.Out(), "Logs:")
+			fmt.Fprintf(dockerCli.Out(), "> => %s:\n", out.Error.Name)
+			for _, l := range out.Error.Logs {
+				fmt.Fprintln(dockerCli.Out(), "> "+l)
 			}
+			fmt.Fprintln(dockerCli.Out())
 		}
-
 		if debug.IsEnabled() {
-			fmt.Fprintf(dockerCli.Out(), "\n%+v\n", stack.Formatter(retErr))
-		} else if len(stack.Traces(retErr)) > 0 {
+			fmt.Fprintf(dockerCli.Out(), "\n%+v\n", stack.Formatter(out.Error.statusErr))
```

Comment:
Could we always have the full stack as well in json output?

Reply:
Yes, don't set the raw error in `out.Error.statusErr`. Instead already split it into separate properties like `Stack` .